src/core/friendly_errors/param_validator.js

```diff
@@ -8,351 +8,366 @@ import { z } from 'zod';
+  //   - Vector: f()
+  // and so on.
+  const p5Constructors = {};
+
+  fn._loadP5Constructors = function () {
+    // Make a list of all p5 classes to be used for argument validation
+    // This must be done only when everything has loaded otherwise we get
+    // an empty array
+    for (let key of Object.keys(p5)) {
+      // Get a list of all constructors in p5. They are functions whose names
+      // start with a capital letter
+      if (typeof p5[key] === 'function' && key[0] !== key[0].toLowerCase()) {
+        p5Constructors[key] = p5[key];
+      }
     }
- * Where each array in `overloads` represents a set of valid overloaded
- * parameters, and `?` is a shorthand for `Optional`.
- *
- * TODO:
- * - [ ] Support for p5 constructors
- * - [ ] Support for more obscure types, such as `lerpPalette` and optional
- * objects in `p5.Geometry.computeNormals()`
- * (see https://github.com/processing/p5.js/pull/7186#discussion_r1724983249)
- *
- * @param {String} func - Name of the function.
- * @returns {z.ZodSchema} Zod schema
- */
-function generateZodSchemasForFunc(func) {
-  // Expect global functions like `sin` and class methods like `p5.Vector.add`
-  const ichDot = func.lastIndexOf('.');
-  const funcName = func.slice(ichDot + 1);
-  const funcClass = func.slice(0, ichDot !== -1 ? ichDot : 0) || 'p5';
-
-  let funcInfo = arrDoc[funcClass][funcName];
+  }
 
-  let overloads = [];
-  if (funcInfo.hasOwnProperty('overloads')) {
-    overloads = funcInfo.overloads;
+  // `constantsMap` maps constants to their values, e.g.
+  // {
+  //   ADD: 'lighter',
+  //   ALT: 18,
+  //   ARROW: 'default',
+  //   AUTO: 'auto',
+  //   ...
+  // }
+  const constantsMap = {};
+  for (const [key, value] of Object.entries(constants)) {
+    constantsMap[key] = value;
   }
 
-  // Returns a schema for a single type, i.e. z.boolean() for `boolean`.
-  const generateTypeSchema = type => {
-    // Type only contains uppercase letters and underscores -> type is a
-    // constant. Note that because we're ultimately interested in the value of
-    // the constant, mapping constants to their values via `constantsMap` is
-    // necessary.
-    if (/^[A-Z_]+$/.test(type)) {
-      return z.literal(constantsMap[type]);
-    } else if (schemaMap[type]) {
-      return schemaMap[type];
-    } else {
-      // TODO: Make this throw an error once more types are supported.
-      console.log(`Warning: Zod schema not found for type '${type}'. Skip mapping`);
-      return undefined;
-    }
+  // Start initializing `schemaMap` with primitive types. `schemaMap` will
+  // eventually contain both primitive types and web API objects.
+  const schemaMap = {
+    'Any': z.any(),
+    'Array': z.array(z.any()),
+    'Boolean': z.boolean(),
+    'Function': z.function(),
+    'Integer': z.number().int(),
+    'Number': z.number(),
+    'Number[]': z.array(z.number()),
+    'Object': z.object({}),
+    // Allows string for any regex
+    'RegExp': z.string(),
```

Comment:
Do we want to also allow RegExp instances here too?

Reply:
What I can find is RegExp is used in `p5.Table.matchRow()`, there may be more places but not many. To check for regex properly it probably should be checking whether the argument is an [instance of `RegExp`](https://zod.dev/?id=instanceof).

Reply:
Replaced string for regexp with the object!